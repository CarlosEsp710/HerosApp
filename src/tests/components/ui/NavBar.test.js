import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";

import { NavBar } from "../../../components/ui/NavBar";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();
const mockDispatch = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("test <NavBar />", () => {
  const authContext = {
    user: {
      isLoggedIn: true,
      name: "Carlos Espejel",
      email: "carlosespejel71@gmail.com",
      imageUrl: "https://avatars.githubusercontent.com/u/57500554?s=96&v=4",
    },
    dispatch: mockDispatch,
  };

  test("should return the component <NavBar />", () => {
    const wrapper = mount(
      <AuthContext.Provider value={authContext}>
        <MemoryRouter initialEntries={["/"]}>
          <NavBar />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("img").at(1).prop("alt")).toBe("Carlos Espejel");
  });

  test("should call logout, navigate and dispatch", () => {
    const wrapper = mount(
      <AuthContext.Provider value={authContext}>
        <MemoryRouter initialEntries={["/"]}>
          <NavBar />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    wrapper.find("button").at(0).simulate("click");

    expect(mockDispatch).toHaveBeenCalled();
    expect(mockDispatch).toHaveBeenCalledWith({ type: types.logout });
    expect(mockNavigate).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledWith("/login", { replace: true });
  });
});
