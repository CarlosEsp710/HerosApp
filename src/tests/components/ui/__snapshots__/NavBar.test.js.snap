// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <NavBar /> should return the component <NavBar /> 1`] = `
<div
  className="min-h-full"
>
  <nav
    className="bg-gray-800"
  >
    <div
      className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        className="flex items-center justify-between h-16"
      >
        <div
          className="flex items-center"
        >
          <div
            className="flex-shrink-0"
          >
            <img
              alt="Workflow"
              className="h-8 w-8"
              src="https://tailwindui.com/img/logos/workflow-mark-indigo-500.svg"
            />
          </div>
          <div
            className="hidden md:block"
          >
            <div
              className="ml-10 flex items-baseline space-x-4"
            >
              <a
                aria-current="page"
                className="bg-gray-900 text-white px-3 py-2 rounded-md text-sm font-medium"
                href="/"
                onClick={[Function]}
              >
                Dashboard
              </a>
              <a
                className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                href="/marvel"
                onClick={[Function]}
              >
                Marvel
              </a>
              <a
                className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                href="/dc"
                onClick={[Function]}
              >
                DC
              </a>
              <a
                className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                href="/search"
                onClick={[Function]}
              >
                Search
              </a>
              <button
                className="hover:bg-gray-100 block px-4 py-2 text-sm text-gray-700"
                onClick={[Function]}
              >
                Logout
              </button>
            </div>
          </div>
          <div
            className="hidden md:block justify-end"
          >
            <div
              className="ml-4 flex items-center md:ml-6"
            >
              <div
                className="ml-3 relative"
              >
                <div>
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="max-w-xs bg-gray-800 rounded-full flex items-center text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                    id="headlessui-menu-button-1"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    type="button"
                  >
                    <span
                      className="sr-only"
                    >
                      Open user menu
                    </span>
                    <img
                      alt="Carlos Espejel"
                      className="h-8 w-8 rounded-full"
                      src="https://avatars.githubusercontent.com/u/57500554?s=96&v=4"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="-mr-2 flex md:hidden"
        >
          <button
            aria-expanded={false}
            className="bg-gray-800 inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
            id="headlessui-disclosure-button-2"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Open main menu
            </span>
            <svg
              aria-hidden="true"
              className="block h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
